accepted/00104-enhanced-ansible-support.md

```diff
@@ -0,0 +1,296 @@
+    ```
+    beacons:
+      inotify:
+        - files:
+            /etc/ansible/hosts:
+              mask:
+                - modify
+            /another/inventory/file:
+              mask:
+               - modify
+        - interval: 5
+        - disable_during_state_run: True
+    ```
+
+    This beacon would be updated whenever a inventory file is added/removed on Uyuni.
+
+    Along with the beacon we'll have to add a new `reactor` to the salt master that will react to events send by the beacons.
+
+    Additionally we'll trigger a manual update of the minion list whenever inventories are added/removed.
+
+2) Calling `ansible.targets` for every inventory regularly can put a lot of extra load on the salt master if there is a greater number of inventories to be scanned.
+
+    To address this we should update the `ansible.targets` function to be able to receive a list of inventories instead of a single one.
+    ```
+    salt 'ansible-control-node.tf.local' ansible.targets inventories='/etc/ansible/hosts, <another_inventory>, <yet_another_one>'
+    ```
+    The output could look something like:
+    ```
+    ansible-control-node.tf.local:
+      /etc/ansible/hosts:
+          ----------
+          Some_group:
+              ----------
+              hosts:
+                  - some-minion.tf.local
+          _meta:
+              ----------
+              hostvars:
+                  ----------
+          all:
+              ----------
+              children:
+                  - Some_group
+                  - ungrouped
+      <another_inventory>:
+          ----------
+          Another_group:
+              ----------
+              hosts:
+                  - another-minion.tf.local
+                  - additional hosts
+          _meta:
+              ----------
+              hostvars:
+                  ----------
+          all:
+              ----------
+              children:
+                  - Another_group
+                  - ungrouped
+      <yet_another_one>:
+        .
+        .
+        .
+    ```
+    This way we'd only have to do one single salt execution per control node. We would then parse the output and compile it into a list of systems. This list will be used to add/remove the `ansible_managed` entitlement from systems registered in Uyuni.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+I don't currently see any drawback with the proposed implementation.
+
+# Future improvements
+- Make Ansible work with Uyunis system groups.
+- Allow adding Ansible playbooks to custom state recurring actions to support executing multiple playbooks in order.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+The following points still need to be investigated in:
+- It looks like the `inotify beacon` requires `python3-inotify package` to be installed on every control node. Is this beacon already available by other means or do we have to make sure the package is installed when setting up a new control node in Uyuni?
```

Comment:
When we set a system as "Ansible Control Node" (in system properties), then this injects some bits to the system highstate, particularly it forces `ansible` package to be installed, so we can also make some other packages to be installed.

But an important thing here is that, beacons would be executed by `venv-salt-minion` (as in 5.0+ all registered clients are migrated to "venv-salt-minion"), so we need python3-inotify for the Salt Bundle. The good thing is that the Salt Bundle already contains `saltbundlepy-inotify`, meaning this will be available in all Salt Bundle supported systems.

Another thing worth to mention in this RFC that beacons won't work in Salt SSH, so this approach won't work for Ansible Control Nodes managed via Salt SSH.